### PR TITLE
Add "trace-macros" as a compiler flag

### PR DESCRIPTION
**Pull request by laumann**
This takes care of #22619

Should the feature and macro be removed as well?

### CODE REVIEW

File: src/librustc/session/config.rs [hunk 1]
```diff
@@ -605,6 +605,8 @@ options! {DebuggingOptions, DebuggingSetter, basic_debugging_options,
           "Print the size of enums and their variants"),
     force_overflow_checks: Option<bool> = (None, parse_opt_bool,
           "Force overflow checks on or off"),
+    trace_macros: bool = (false, parse_bool,
+          "Trace all macro invocations"),
```
Comment (nrc) [inline src/librustc/session/config.rs]: Could this be a bit more descriptive? What is the user-visible effect of tracing?

**Comment by nrc** [comment]
The code looks good. Could you add a test please? r+ with that.

Ping @kmcallister  on the question of whether to remove the existing feature/macro.

**Comment by laumann** [comment]
@nrc yes, I'll add a test
@kmcallister should the existing feature/macro be removed?

**Comment by laumann** [comment]
Closing this for now - I seem to have messed it up somewhat, and I will not be able to work on it for the next week.

**Pull request closed without merging**